src/common/wayfinder/verification/trusted-gateway.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DataHashProvider, DataVerifier } from '../../../types/wayfinder.js';
+
+const arioGatewayHeaders = {
+  digest: 'x-ar-io-digest',
+  verified: 'x-ar-io-verified',
+};
+
+export class TrustedGatewayHashProvider implements DataHashProvider {
```

Comment:
Nit: consider making this plural - `TrustedGatewaysHashProvider`.

Reply:
[d9ca652](https://github.com/ar-io/ar-io-sdk/pull/466/commits/d9ca652d1ed0b4b2ee553f7f148d6738ca0dbbf3)